### PR TITLE
[16.0][IMP] purchase_sale_container_deposit_inter_company: Allow update so lines on locked sale when update order container deposit quantity

### PR DESCRIPTION
**Pull request by QuocDuong1306**
Depend on:
- https://github.com/OCA/multi-company/pull/533
- https://github.com/OCA/multi-company/pull/602

### CODE REVIEW

File: purchase_sale_container_deposit_inter_company/models/purchase_order.py [hunk 1]
```diff
@@ -20,3 +20,9 @@ def _prepare_sale_order_line_data(self, purchase_line, dest_company, sale_order)
         # Copy value from PO to SO
         order_line["is_container_deposit"] = purchase_line.is_container_deposit
         return order_line
+
+    def write(self, vals):
+        # Allow update of so lines on locked sale on update of order container deposit quantity
+        if self.env.context.get("update_order_container_deposit_quantity", False):
```
Comment (santostelmo) [inline purchase_sale_container_deposit_inter_company/models/purchase_order.py:26]: Can we have a test that check that we are able to update locked sales while we are updating order container deposit ?

Comment (QuocDuong1306) [reply purchase_sale_container_deposit_inter_company/models/purchase_order.py:26]: Hello @santostelmo , I updated

**Review by santostelmo — APPROVED** [review]

**Review by gurneyalex — APPROVED** [review]

**Comment by QuocDuong1306** [comment]
Hi @dreispt , the PR is updated, could you have a look and merge it?

**Review by simahawk — APPROVED** [review]

**Pull request merged**